Problem1/Problem1/src/problem1/javaWindow.java

```diff
@@ -0,0 +1,269 @@
+
+    }//GEN-LAST:event_decimalOutputActionPerformed
+    /**
+     * When clicked it will convert binary to decimal
+     *
+     * @param evt
+     */
+    private void convertToDecimalActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_convertToDecimalActionPerformed
+
+        try {
+            if (!isBinary(binaryInput.getText())) {
+                Toolkit.getDefaultToolkit().beep();
+                decimalOutput.setText("Invalid input");
+            } else {
+                int input = Integer.parseInt(binaryInput.getText());
+                decimalOutput.setText(convert.toDecimal(input));
+            }
+        } catch (NumberFormatException nfe) {
+            Toolkit.getDefaultToolkit().beep();
+            decimalOutput.setText("Invalid input");
+            nfe.printStackTrace(System.out);
+        }
+    }//GEN-LAST:event_convertToDecimalActionPerformed
+
+    /**
+     * @param args the command line arguments
+     */
+    public static void main(String args[]) {
+        /* Set the Nimbus look and feel */
+        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
+        /* If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel.
+         * For details see http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html 
+         */
+        try {
+            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
+                if ("Nimbus".equals(info.getName())) {
+                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
+                    break;
+                }
+            }
+        } catch (ClassNotFoundException ex) {
+            java.util.logging.Logger.getLogger(javaWindow.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
+        } catch (InstantiationException ex) {
+            java.util.logging.Logger.getLogger(javaWindow.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
+        } catch (IllegalAccessException ex) {
+            java.util.logging.Logger.getLogger(javaWindow.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
+        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
+            java.util.logging.Logger.getLogger(javaWindow.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
+        }
+        //</editor-fold>
+        //</editor-fold>
+
+        java.awt.EventQueue.invokeLater(new Runnable() {
+            @Override
+            public void run() {
+                new javaWindow().setVisible(true);
+            }
+        });
+    }
+
+    /**
+     * Checks if binary or not
+     *
+     * @param input
+     * @return boolean
+     */
+    public static boolean isBinary(String input) {
+        for (int i = 0; i < input.length(); i++) {
+            if (input.charAt(i) != '0' || input.charAt(i) != '1') {
```

Comment:
Your “if” statement logic is wrong ( the || should be &&).  It is detecting all numbers as invalid.